src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -464,6 +469,11 @@ internal void ProcessPendingAck(TdsParserStateObject stateObj)
                 _physicalStateObj.AssignPendingDNSInfo(serverInfo.UserProtocol, FQDNforDNSCahce, ref _connHandler.pendingSQLDNSObject);
             }
 
+            if (encrypt && !EncryptClientPossible)
+            {
+                _encryptionOption = EncryptionOptions.NOT_SUP;
```

Comment:
1. _encryptionOption should be set to NOT_SUP even when "encrypt" is false.
2. If "encrypt" is true instead of setting the option value to NOT_SUP, can you throw below error directly and terminate?
https://github.com/dotnet/SqlClient/pull/1210/files#diff-9468409fbd86dc6afecde79621ca15d128de0597cb0d36ce8464b469bc62a5b8R690

Reply:
If the idea of throwing an exception even by encrypt=false is approved we can raise the exception here as you mentioned.

Reply:
No I meant:

```
if (!EncryptClientPossible)
{
    if(encrypt)
    {
        throw exception;
    }
    _encryptionOption = EncryptionOptions.NOT_SUP;
}
```

Reply:
If we're good by throwing an exception on either encrypt is `false` or `true` with the message "The instance of SQL Server you attempted to connect to requires encryption but this machine does not support it.", it's possible to handle those conditions here.

Reply:
As discussed offline, please update design to respect specifications for encryption options supported by driver: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-tds/60f56408-0188-4cd5-8b90-25c6f2423868